### PR TITLE
Lesson 1.7

### PR DESCRIPTION
**Pull request by larasodjati**
Removed custom hook, added asynchronous operation into the app component, added loading state, and created conditional loading indicator.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,34 +2,75 @@ import React from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-const useSemiPersistentState = () => {
-  const [todoList, setTodoList] = React.useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
+// const useSemiPersistentState = () => {
```
Comment (Atos20) [inline src/App.js]: let's remove this commented code.

File: src/App.js [hunk 2]
```diff
@@ -2,34 +2,75 @@ import React from 'react';
...
+//   return [todoList, setTodoList];
+// };
 
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  React.useEffect(() => {
+    new Promise((resolve, reject) => {
```
Comment (Atos20) [inline src/App.js]: my only suggestion is to extract the `new Promise` to its own functions to make it cleaner.

**Review by Atos20 — APPROVED** [review]
Great job!

File: src/App.js (hunk 2 above)
Comment (larasodjati) [reply src/App.js]: @Atos20 Noted, thanks for the feedback. I created a new function for this new Promise and pushed it to the remote branch.

**Pull request merged**